### ISSUE TITLE
Maven project with dependent jar: Unable to set extraClasspath to file

### ISSUE THREAD

**Issue by christoforosl** [HotswapProjects/HotswapAgent#101]
Setup: 
Maven parent pom with 2 projects: war (Web Application) and dependent jar in directory DTPHModel.  
Server: Jboss -eap-6.4 running under JDK 1.8.0_45 with hotswap-agent.jar version 0.3.0-SNAPSHOT

JBOSS startup values: 
```
-Xms128m -Xmx1g -XXaltjvm=dcevm -javaagent:c:\progra~1\NetBeans8.0.1\hotswap-agent.jar -Xdebug -Xnoagent -Djava.compiler=NONE -Xrunjdwp:transport=dt_socket,address=8787,server=y,suspend=n"
```

hotswap-agent.properties, in Web Application's src/main/resources has the following entries:

```
extraClasspath=${basedir}/../DTPHModel/target/classes
 # ${basedir} will be replaced by Maven <resource><filtering>true</filtering>.. option
autoHotswap=true
autoHotswap.port=8000
LOGGER=debug
```

The aim here is to hot reload classes in DTPHModel/target/classes directory.  When the web application starts I get error 
```
13:35:35,919 INFO  [stdout] (Thread-207) HOTSWAP AGENT: 13:35:35.919 DEBUG (org.hotswap.agent.config.PluginConfiguration) - Unable to set extraClasspath to [file:/C:/Users/christoforosl/java/projects/poleodomia/javaApplications/DTPHModel/target/classes/] on classLoader ModuleClassLoader for Module "deployment.CentralManagement.war:main" from Service Module Loader. Only URLClassLoader is supported.
13:35:35,921 INFO  [stdout] (Thread-207) *** extraClasspath configuration property will not be handled on JVM level ***
```

Problem is,  [file:/C:/javaApplications/DTPHModel/target/classes/]
does exist!

**Comment by skybber** [comment]
Jboss ModuleClassLoader is unfortunately not  supported by HotswapAgent now.

**Comment by skybber** [comment]
There is a new patch for jboss-modules in the repository. It could solve your problem,

**Comment by allstone** [comment]
is ModuleClassLoader already supported?

On Thu, Nov 26, 2015 at 8:01 PM, skybber <notifications@github.com> wrote:

> There is a new patch for jboss-modules in the repository. It could solve
> your problem,
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/HotswapProjects/HotswapAgent/issues/101#issuecomment-159970182>
> .
>



-- 
Pagarbiai,
Andrius Stašauskas

**Comment by skybber** [comment]
Yes, but is it still alpha version. I'll appreciate any feedback.

**Comment by christoforosl** [comment]
I downloaded and build latest code, and tested with Jboss -eap-6.4 running under Windows 11 and JDK 1.8.0_45.
I have a war web app which serves JSON responses through REST API calls. I also have dependency on a jar which contains all Model Objects, basically POJOs.  No Spring/Hibernate in the picture.  IDE is Netbeans 8.0.2

hotswap-agent.properties, in Web Application's src/main/resources has the following entries:
```
extraClasspath=${basedir}\\..\\DTPHModel\\target\\classes
autoHotswap=true
LOGGER=debug
```
In the jar dependency, I have a POJO that is serialized as JSON and returned to the client.  I am modifying properties of the POJO in the IDE, save the java file, then observe the JSON Serialization output from a browser.

Results:
1. Modified return values (not datatype) of methods: OK
2. Modified datatype return values of methods: (ie from "public String" to "public Integer" java.lang.InternalError: invoke, redeployment required or error to go away
3. Modified method name: java.lang.InternalError: invoke, redeployment required or error to go away
4. Added method name: No exception thrown, but new method not getting picked up by serializer.
5. Removed method name:  java.lang.InternalError: invoke, redeployment required or error to go away

Also, I was periodically getting jvm exception:
```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  EXCEPTION_ACCESS_VIOLATION (0xc0000005) at pc=0x000000007780c2fc, pid=2604, tid=1360
#
# JRE version: Java(TM) SE Runtime Environment (8.0_45-b15) (build 1.8.0_45-b15)
# Java VM: Dynamic Code Evolution 64-Bit Server VM (25.51-b03-dcevmlight-3 mixed mode windows-amd64 compressed oops)
# Problematic frame:
# V  [jvm.dll+0x1ac2fc]
#
# Failed to write core dump. Minidumps are not enabled by default on client versions of Windows
#
# An error report file with more information is saved as:
# C:\Users\chris\java\jboss-eap-6.4\bin\hs_err_pid2604.log
#
# If you would like to submit a bug report, please visit:
#   https://github.com/dcevm/dcevm/issues
# 
```
If you need hs_err_pid2604.log, I can sent.
Hope this helps.

**Comment by skybber** [comment]
Thanks. As for "java.lang.InternalError" it looks there is a cache that should be cleared. The easiest way for me to fix it is to have an example. Could you please make a minimal example project and send me it via email? It would help a lot to make advance in the jbosss-modules plugin. As for dcevm error there can be a lot of causes. For example I've some deserialization runtime errors after change class fields (then  serialVersionUID should be changed before class swap). Send me please hs_err_pid2604.log as well. (My email is in "git log")
